src/Sentry/Laravel/Features/QueueIntegration.php

```diff
@@ -62,6 +68,43 @@ public function onBoot(Dispatcher $events): void
         }
     }
 
+    public function handleJobQueueingEvent(JobQueueing $event): void
+    {
+        $parentSpan = SentrySdk::getCurrentHub()->getSpan();
+
+        // If there is no tracing span active there is no need to handle the event
+        if ($parentSpan === null) {
+            return;
+        }
+
+        $jobName = $event->job;
+
+        if ($jobName instanceof Closure) {
+            $jobName = 'Closure';
+        } elseif (is_object($jobName)) {
+            $jobName = get_class($jobName);
+        }
+
+        $context = (new SpanContext)
+            ->setOp('queue.publish')
+            ->setData([
+                'job' => $jobName,
+                'connection' => $event->connectionName,
```

Comment:
Would be nice if we can fill in `messaging.system` from https://opentelemetry.io/docs/specs/semconv/attributes-registry/messaging/

Otherwise we probably want to use `messaging.destination.name` instead of `job` and maybe find the OTEL equivalent for `connectionName`? If we can't find anything for `connectionName` I'm fine to keep it.

Reply:
`messaging.system` doesn't have a well-known one for Laravel but I can set it to `laravel` which seems to be allowed.

The `job` is the name of the handler (usually a class) and not the name of the queue it's published on, does that still make sense to use `messaging.destination.name` for that? Eventually I'd like to add the actual queue name too but we need some changes on the framework side to surface that and we might want to reserve the destination name for that info.

Maybe the connection could be in `messaging.destination.connection` (not-standard). In Laravel you can have multiple queue connections configured like `sqs` or `redis` or a fake `sync` driver and the connection name indicates which queue connection it was pushed to.

To give an example of how a Laravel queue message might look like:

```
job: App\Jobs\SendSomeEmail
queue: mailer
payload: <JSON>
connectionName: redis
```

Reply:
> The job is the name of the handler (usually a class) and not the name of the queue it's published on, does that still make sense to use messaging.destination.name for that

Ahhh okay, then maybe we namespace to `laravel` to have it be less generic?

> Maybe the connection could be in messaging.destination.connection

Makes sense to me, let's do that change.

Reply:
Okay, I've made those changes, the op set to `queue.publish` is good?